rfc/rfc-89/rfc-89.md

```diff
@@ -0,0 +1,348 @@
+Here, the parser is designed to be extensible, allowing users to customize parsers.
+
+#### Demo
+```sql
+CREATE TABLE hudi_table(
+  id BIGINT,
+  name STRING,
+  price DOUBLE
+)
+WITH (
+'connector' = 'hudi',
+'path' = 'file:///tmp/hudi_table',
+'table.type' = 'MERGE_ON_READ',
+'index.type' = 'BUCKET',
+'hoodie.bucket.index.partition.expressions' = '\\d{4}-(06-(01|17|18)|11-(01|10|11)),256',
+'hoodie.bucket.index.num.buckets' = '10'
+);
+```
+
+For the dates of 06-01, 06-17, 06-18 in June and 01-11, 10-11, 11-11 in
+November of each year (in the format of yyyy-MM-dd), the corresponding bucket number for the partition is 256
+
+For common partitions use 10 as partition bucket number
+
+
+#### Config Management
+
+The expression config will be persisted as files named as `<instant>.hashing_config` for current table, it stores in
+`.hoodie/.simple_hashing/config` directory and contains the following information in json format
+
+```json
+{
+    "<instant1>": {
+       "expression": "expression1,bucket-number1;expression2,bucket-number2",
+       "rule": "rule-engine",
+       "default": "default-bucket-number"
+    },
+    "<instant2>": {}
+}
+```
+Expressions has newer instant has higher priority
+
+For a specific expression like instant1 ==> Expression1 has a higher priority than expression2.
+
+Here, the config is designed to support multiple versions. During the DDL phase, hoodie will create a `00000000000000000.hashing_config` for the first time. 
+Subsequently, when modifying the expression through the provided `call command`(details refer to Tool ), 
+a new version of the config will be generated. The new-version config contains all historical change records. During 
+loading stage, the latest completed instant related config will be loaded based on the active timeline.
+
+```text
+.hoodie/.simple_hashing/config/00000000000000000.hashing_config
+.hoodie/.simple_hashing/config/20250303095546020.hashing_config
+```
+
+Using Archive Service to clean up expired config versions, keeping up to 5 historical versions
+
+### Hashing Metadata
+Here, the Hashing Metadata is designed to support multiple versions, new versions contains all the info in old versions.
+
+The hashing metadata will be persisted as files named as `<instant>.hashing_meta` for current table, it stores in
+`.hoodie/.simple_hashing/meta` directory and contains the following information
+
+```avro schema
+{
+   "namespace":"org.apache.hudi.avro.model",
+   "type":"record",
+   "doc": "hashing meta for current table using partition level simple bucket index",
+   "name":"SimpleHashMeta",
+   "fields":[
+     {
+       "name":"version",
+       "type":["int", "null"],
+       "default": 1
+     },
+     {
+       "name": "partitionMeta",
+       "type": [
+         "null", {
+           "type":"map",
+           "values": {
+             "type": "record",
+             "name": "PartitionHashInfo",
+             "fields": [
+               {
+                 "name": "bucketNumber",
```

Comment:
we already have `<instant>.hashing_config`, isn't it can be utilitized to infer the partition buckets number already?